eva-accession-clustering-automation/cluster_multiple_assemblies.py

```diff
@@ -0,0 +1,120 @@
+
+
+def preliminary_check(source, asm_vcf_prj_list, assembly_list):
+    """
+    This checks must pass in order to run the script
+    """
+    check_valid_sources(source)
+    check_requirements(source, asm_vcf_prj_list, assembly_list)
+
+
+def cluster_multiple_from_mongo(source, assembly_list, private_config_file, private_config_xml_file, profile,
+                                output_directory, clustering_artifact, only_printing):
+    """
+    This method splits the list of assemblies and call the run_clustering method for each assembly
+    """
+    for assembly in assembly_list.split(','):
+        run_clustering(source, None, None, assembly, private_config_file, private_config_xml_file, profile,
+                       output_directory, clustering_artifact, only_printing, timestamp)
+
+
+def cluster_multiple_from_vcf(source, asm_vcf_prj_list, private_config_file, private_config_xml_file, profile,
+                              output_directory, clustering_artifact, only_printing):
+    """
+    The list will be of the form: asm1,GCA000000001.1,PRJEB1111#asm2,GCA000000002.2,PRJEB2222...
+    This method splits the triplets and then call the run_clustering method for each one
+    """
+    for triplet in asm_vcf_prj_list.split(','):
+        data = triplet.split('#')
+        run_clustering(source, data[1], data[2], data[0], private_config_file, private_config_xml_file, profile,
+                       output_directory, clustering_artifact, only_printing, timestamp)
+
+
+def check_requirements(source, asm_vcf_prj_list, assembly_list):
+    """
+    This method checks depending on the source, what list should have been provided.
+    For VCF it is expected to have a list of one or more assembly, vcf file, project separated by #
+    For Mongo it is expected to have a list of assemblies.
+    """
+    if source.upper() == 'VCF' and not asm_vcf_prj_list:
+        logger.error('If the source is VCF a list of assembly#vcf#project mus be provided using the parameter'
+                     '--asm-vcf-prj-list')
+        sys.exit(1)
+    if source.upper() == 'MONGO' and not assembly_list:
+        logger.error('If the source is MONGO a list of assembly accessions must be provided using the parameters '
+                     '-al or --assembly-list')
+        sys.exit(1)
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description='Cluster multiple assemblies', add_help=False)
+    parser.add_argument("--source", help="mongo database or VCF", required=True)
+    parser.add_argument("--asm-vcf-prj-list", help="List of Assembly, VCF, project to be clustered, "
+                                                   "e.g. GCA_000233375.4#/nfs/eva/accessioned.vcf.gz#PRJEB1111, "
+                                                   "GCA_000002285.2#/nfs/eva/file.vcf.gz#PRJEB2222. "
+                                                   "Required when the source is VCF",
+                        required=False)
```

Comment:
You do not need to use coma separated values, you can use `nargs='+'` 
```suggestion
    parser.add_argument("--asm-vcf-prj-list", help="List of Assembly, VCF, project to be clustered, "
                                                   "e.g. GCA_000233375.4#/nfs/eva/accessioned.vcf.gz#PRJEB1111  "
                                                   "GCA_000002285.2#/nfs/eva/file.vcf.gz#PRJEB2222. "
                                                   "Required when the source is VCF",
                        required=False, nargs='+')
```

Reply:
Thanks, done